src/drivers/actuators/vertiq_io/vertiq_io.hpp

```diff
@@ -0,0 +1,193 @@
+
+	int print_status() override;
+
+	/** @see ModuleBase::run() */ //I do not think this actually comes from ModuleBase. it should come from scheduled work item
+	void Run() override;
+
+	/** @see OutputModuleInterface */
+	bool updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS],
+			   unsigned num_outputs, unsigned num_control_groups_updated) override;
+
+private:
+	static const uint8_t MAX_SUPPORTABLE_IFCI_CVS = 16;
+
+	enum DISARM_BEHAVIORS {TRIGGER_MOTOR_DISARM, COAST_MOTOR, SEND_PREDEFINED_THROTTLE};
+	enum ARM_BEHAVIORS {USE_MOTOR_ARMING, FORCE_ARMING};
+
+	//Variables and functions necessary for properly configuring the serial interface
+	//Determines whether or not we should initialize or re-initialize the serial connection
+	static px4::atomic_bool _request_telemetry_init;
+
+	MixingOutput _mixing_output{"VERTIQ_IO", MAX_SUPPORTABLE_IFCI_CVS, *this, MixingOutput::SchedulingPolicy::Auto, false, false};
+
+	//The name of the device we're connecting to. this will be something like /dev/ttyS3
+	static char _telemetry_device[20];
+
+	//Counters/timers to track our status
+	perf_counter_t	_loop_perf{perf_alloc(PC_ELAPSED, MODULE_NAME": cycle")};
+	perf_counter_t	_loop_interval_perf{perf_alloc(PC_INTERVAL, MODULE_NAME": output update interval")};
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface _serial_interface;
+
+	//We need someone who can manage our clients
+	VertiqClientManager _client_manager;
+
+	//We need a telemetry handler
+	VertiqTelemetryManager _telem_manager;
+
+	//Pointer to the IFCI handler
+	IFCI *_motor_interface_ptr;
+
+	//Store the number of control variables that we're using
+	uint8_t _cvs_in_use = 0;
+
+	//Store the telemetry bitmask for who we want to get telemetry from
+	uint16_t _telem_bitmask = 0;
+
+	//This is the variable we're actually going to use in the brodcast packed control message
+	//We set and use it to _current_telemetry_target_module_id until we send the first
+	//broadcast message with this as the tail byte. after that first transmission, set it
+	//to an impossible module ID
+	uint16_t _telemetry_request_id = 0;
+
+	static const uint8_t _impossible_module_id = 255;
+
+	bool _send_forced_arm = true;
+	bool _actuator_test_active = false;
+
+	//We want to publish our ESC Status to anyone who will listen
+	uORB::Publication<esc_status_s> _esc_status_pub{ORB_ID(esc_status)};
+
+	//We need to know what's going on with the actuator test to make sure we handle it properly
+	uORB::Subscription _actuator_test_sub{ORB_ID(actuator_test)};
+	actuator_test_s _actuator_test;
+
+	/**
+	 * Check for parameter changes and update them if needed.
+	 */
+	void parameters_update();
+
+	// Subscriptions
+	uORB::SubscriptionInterval _parameter_update_sub{ORB_ID(parameter_update), 1_s};
+
+	//We need to bring in the parameters that we define in module.yaml in order to view them in the
+	//control station, as well as to use them in the firmware
+	DEFINE_PARAMETERS(
```

Comment:
these are the params that will get out into the control station. Users can add whatever they want to this if they want to expose a new param.